dev/triggers.h

```diff
@@ -0,0 +1,204 @@
+#pragma once
+
+#include <memory>
+#include <sstream>
+#include <string>
+#include <tuple>
+
+#include "tuple_helper/tuple_helper.h"
+
+// NOTE Idea : Maybe also implement a custom trigger system to call a c++ callback when a trigger triggers ?
+// (Could be implemented with a normal trigger that insert or update an internal table and then retreive
+// the event in the C++ code, to call the C++ user callback, with update hooks: https://www.sqlite.org/c3ref/update_hook.html)
+// It could be an interesting feature to bring to sqlite_orm that other libraries don't have ?
```

Comment:
You can't do it in pure SQL.
But with C, there is https://www.sqlite.org/c3ref/update_hook.html so I think it may be possible to use the update hook (which is basicly a callback when a update/delete/insert event occurs on a table).

Although, there is one drawback: the callback cannot do anything with the database (no querry while in the callback function). (we could maybe allow the user to use the DB in his callback, but it will require more work to defere the call of the user callback for after we return from the real callback)

I don't know if that's really an intresting feature, it was just a thought I had when I saw that function in Sqlite's documentation

Reply:
It is an interesting thought. Anyway it is a good idea to implement update hook and see whether anyone request the feature you are talking about. I'll add update hook inside todo list